mtgjson5/cache_builder.py

```diff
@@ -0,0 +1,429 @@
+            .collect()
+        )
+        
+    def _load_uuid_cache(self) -> None:
+        """Load legacy UUID mappings if available."""
+        cache_path = constants.RESOURCE_PATH / "legacy_mtgjson_v5_uuid_mapping.json"
+        if not cache_path.exists():
+            LOGGER.info("No UUID cache found")
+            return
+        try:
+            self.uuid_cache_df = pl.read_json(cache_path)
+            LOGGER.info(f"Loaded UUID cache with {len(self.uuid_cache_df)} entries")
+        except Exception as e:
+            LOGGER.warning(f"Failed to load UUID cache: {e}")
+            self.uuid_cache_df = None
+    
+    def _load_external_providers(self) -> None:
+        """Load external provider data as DataFrames."""
+        ck_data = self.card_kingdom.download(self.card_kingdom.api_url)
+        self.card_kingdom_df = pl.DataFrame(ck_data.get("data") or [])
+        self.card_kingdom_map = self.card_kingdom.get_scryfall_translation_table()
+        self.card_market_provider = CardMarketProvider()._get_card_mmarket_data()
+        self.multiverse_bridge_map = (
+            MultiverseBridgeProvider().get_rosetta_stone_cards()
+        )
+        self.cardhoarder_map = CardMarketProvider().get_cardmarket_price_map()
+
+        self.uuid_cache_df = self._load_uuid_cache()
+
+    def _load_uuid_cache(self) -> pl.DataFrame | None:
+        """Load legacy UUID mappings if available."""
+        cache_path = Path("resources/legacy_mtgjson_v5_uuid_mapping.json")
+        if not cache_path.exists():
+            return None
+        return pl.read_json(cache_path)
+
+    def get_set(self, set_code: str) -> dict[str, Any] | None:
+        """Get set metadata by set code."""
+        return self.sets_map.get(set_code.upper())
+
+    def get_foreign_data(
+        self, set_code: str, collector_number: str
+    ) -> list[dict[str, Any]]:
+        """Get foreign language data for a specific card."""
+        if self.foreign_data_map is None:
+            return []
+
+        result = self.foreign_data_map.filter(
+            (pl.col("set") == set_code.upper())
+            & (pl.col("collector_number") == collector_number)
+        )
+
+        if result.is_empty():
+            return []
+
+        # Extract and force to Python list
+        foreign_data = result["foreign_data"].to_list()[0]
+
+        if foreign_data is None:
+            return []
+
+        return list(foreign_data)
+
+    def get_meld_parts(self, result_name: str) -> list[str] | None:
+        """Get meld part names for a meld result card."""
+        for triplet in self.meld_triplets:
+            if triplet[2] == result_name:
+                return triplet[:2]
+        return None
+
+    @classmethod
+    def get_instance(cls) -> "GlobalCache":
+        """Get the singleton instance of GlobalCache."""
+        if cls._instance is None:
+            cls()
+        assert cls._instance is not None
+        return cls._instance
+
+
+global_cache = GlobalCache.get_instance()
```

Comment:
Can we call this GLOBAL_CACHE instead? Also, how does this interact with the current cache scheme?